helix-view/src/document.rs

```diff
@@ -291,8 +378,58 @@ impl Document {
         }
     }
 
+    pub(crate) async fn save_impl<'a>(
```

Comment:
I suspect I'm just out of the loop on this, and there's a reason this needs to be async right now.

But if it doesn't need to be, I'd say remove the async for now, and we can add it for both `open()` and `save()` when we actually have an application for it.

Reply:
It's because you can't really pass an async writer to a sync function.